src/runtime/runtime_nintendoswitch.go

```diff
@@ -0,0 +1,75 @@
+// +build nintendoswitch
+
+package runtime
+
+type timeUnit int64
+
+const asyncScheduler = false
+
+func postinit() {}
+
+// Entry point for Go. Initialize all packages and call main.main().
+//export main
+func main() int {
+	preinit()
+	run()
+
+	return exit(0) // Call libc_exit to cleanup libnx
+}
+
+// sleepTicks argument are actually in microseconds
+func sleepTicks(d timeUnit) {
+	usleep(uint(d))
```

Comment:
If I understand correctly:

* `armGetSystemTick` returns the time in 19200000Hz units.
* `usleep` (as the name indicates) sleeps in microseconds.

The `timeUnit` type assumes there is just one unit, not two. In fact, in the implementation here values of `timeUnit` can be either in nanoseconds or in the Switch-specific time unit (with intervals of around 52ns). This is not consistent with other targets, where it means just one thing.

I would suggest switching to nanoseconds everywhere for consistency. The conversion to/from microseconds is obvious. The conversion to/from the system tick timer is less obvious, and assuming 52ns tick interval is slightly incorrect. You can use this instead, which is exact:

```go
func ticks() timeUnit {
	// the following formula is equivalent to this but with less risk of overflowing:
	//     getArmSystemTimeNs() * 1e9 / 19200000
	return getArmSystemTimeNs() * 625 / 12
}
```

Reply:
I think perhaps #1127 is the model to look at now, of how to do this.